GrobExp.Compiler/ExpressionEmitters/CallExpressionEmitter.cs

```diff
@@ -18,21 +19,25 @@ protected override bool EmitInternal(MethodCallExpression node, EmittingContext
             var method = node.Method;
             Expression obj;
             IEnumerable<Expression> arguments;
+            IEnumerable<ParameterInfo> parameters;
             bool isStatic = method.IsStatic;
             if(!isStatic)
```

Comment:
I think that this code will become more readable if we write all CallExpression emitting logic separately for three cases: a) non-static methods, static extension methods on IEnumerabe<T> with "treat nulls as empty arrays" tweak, and c) all other static methods.